cirq/ops/common_gates.py

```diff
@@ -322,6 +332,29 @@ def _act_on_(self, args: Any):
                                                         tableau.xs[:, q].copy())
             return True
 
+        if isinstance(args, clifford.ActOnStabilizerCHFormArgs):
+            if protocols.is_parameterized(self) or self.exponent % 0.5 != 0:
+                return NotImplemented
+            effective_exponent = self._exponent % 2
+            state = args.state
+            if effective_exponent == 0.5:
+                assert all(
+                    gate._act_on_(args)  # type: ignore
+                    for gate in [ZPowGate(), HPowGate()])
+                state.omega *= (1 + 1j) / (2**0.5)  # type: ignore
```

Comment:
Why is the type ignore needed?

Reply:
mypy assumes state.omega is an int since it was initialized by an int. It complains when I try to assign a complex value to it. Maybe adding `+ 0j` to the initialization will fix it. Let me know if you think that's better.